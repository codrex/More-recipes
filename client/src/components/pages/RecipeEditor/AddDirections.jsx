import React from 'react';
import { reduxForm } from 'redux-form';
import { connect } from 'react-redux';
import AddItems from './AddItems';
import {
  updateDirections,
  clearValidationError
} from '../../../actions/recipeActions';

/**
 * AddDirections component
 * @param {object} props
 * @return {React} react element
 */
const AddDirections = props => (
  <AddItems
    name="direction"
    directions
    placeholder="Enter direction"
    {...props}
  />
);

const mapStateToProps = state => (
  {
    items: state.recipe.directions,
    externalError: state.recipeValidationError.directions,
    message: state.networkRequest.msg
  }
);

export { AddDirections as PureAddDirections };

export default reduxForm({
  form: 'directionForm'
})(connect(mapStateToProps, {
  sendItemsToStore: updateDirections,
  clearValidationError
})(AddDirections));
