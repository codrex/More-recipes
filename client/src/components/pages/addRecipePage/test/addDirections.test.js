import React from 'react';
import AddDirections from '../addDirections/addDirections';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../../../../store/configStore';

const props = {
  clearValidationError: jest.fn(),
  directions: true,
  externalError: [],
  handleSubmit: jest.fn(),
  ingredients: false,
  initialize: jest.fn(),
  items: [],
  name: '',
  placeholder: '',
  sendItemsToStore: jest.fn(),
};
test('Add directions component to render as expected', () => {
  const component = renderer.create(
    <Provider store={configureStore()}>
      <AddDirections />
    </Provider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

