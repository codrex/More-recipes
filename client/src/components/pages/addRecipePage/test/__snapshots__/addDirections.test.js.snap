// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add directions component to render as expected 1`] = `
<div
  className="col-12 items "
>
  <div
    className="items-header"
  >
    <h4
      className="items-header-text"
    >
      directions  0
    </h4>
    <form
      action=""
      className="col-12"
      method=""
      onSubmit={[Function]}
    >
      <div
        className="form-group d-flex flex-column no-margin"
      >
        <textarea
          aria-describedby="directionTextarea"
          className="form-control text-area no-margin add-item-input"
          id="directionTextarea"
          name="direction"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Enter direction"
          rows="1"
          value=""
        />
        <div
          className=""
        />
      </div>
      <button
        className="btn text-capitalize w-100 btn-secondary"
        disabled={false}
        id="submit"
        type="submit"
      >
        add direction
      </button>
    </form>
  </div>
  <div
    className="col-12 items-list-wrapper"
  >
    <div />
  </div>
</div>
`;
